lib/oca-epak/base_client.rb

```diff
@@ -13,24 +13,48 @@ def initialize(username, password)
 
     protected
 
-      def parse_result(response, method)
-        method_response = "#{method}_response".to_sym
-        method_result = "#{method}_result".to_sym
-        if body = response.body[method_response]
-          body[method_result]
-        end
-      end
+    def parse_result(response, method)
+      schema = parse_body(response, method)[:schema]
+      payload = parse_body(response, method)[:diffgram]
+      results_key, results_topic_keys = get_result_keys(schema)
 
-      # @return [Array, nil]
-      def parse_results_table(response, method)
-        if result = parse_result(response, method)
-          if result[:diffgram][:new_data_set]
-            table = result[:diffgram][:new_data_set][:table]
-            table.is_a?(Hash) ? [table] : table
-          else
-            raise Oca::Errors::BadRequest.new("Oca WS responded with:\n#{response.body}")
+      if results_key == :errores
+        error = payload[results_key][results_topic_keys.first]
+        raise Oca::Errors::BadRequest.new("Oca WS responded with:\n#{error.description}")
+      elsif body = payload[results_key]
```

Comment:
@cavi21 This `elsif` doesn't have an `else` - considering that the response is coming from OCA, shouldn't we have a default `else` that raises something like `Oca::Errors::BadResponse` or something like that?